File: algorithms/Strings/Easy/FirstUniqChar/firstUniqChar.py
# Given a string s, find the first non-repeating character in it and return its index. If it does not exist, return -1.

 

# Example 1:

# Input: s = "leetcode"
# Output: 0
# Example 2:

# Input: s = "loveleetcode"
# Output: 2
# Example 3:

# Input: s = "aabb"
# Output: -1
 

# Constraints:

# 1 <= s.length <= 105
# s consists of only lowercase English letters.


def firstUniqChar(string :str)->int:
    hashmap=dict()
    for index in range(len(string)):
        if string[index] in hashmap.keys():
            hashmap[string[index]]+=1
        else:
            hashmap[string[index]]=1
    
    for index in range(len(string)):
        if hashmap[string[index]]==1:
            return index 
    return -1

File: algorithms/Strings/Easy/FirstUniqChar/firstUniqChar.cpp
#include<iostream>
#include<bits/stdc++.h>
#include<string>


using namespace std;

int firstuniqchar(string s){
    map<char,int> hashmap;

    for(int i=0;i<s.size();i++){
        char ch=s[i];
        if(hashmap.find(ch)==hashmap.end()){
            hashmap[ch]=1;
        }
        else{
            hashmap[ch]+=1;
        }


    }
    for(int i=0;i<s.size();i++){
        if(hashmap[s[i]]==1)
        return i;
    }
    return -1;
}

int main(){

    int ans=firstuniqchar("vsdevelopers");
    cout<<ans;
    return 0;
}